pkg/controller/servicebindingrequest/annotations/annotations.go

```diff
@@ -0,0 +1,72 @@
+package annotations
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+)
+
+// bindingType encodes the medium the binding should deliver the configuration value.
+type bindingType string
+
+const (
```

Comment:
is this a good opportunity to switch to the [new annotation data model](https://github.com/application-stacks/service-binding-specification/blob/master/annotations.md#data-model--building-blocks-for-expressing-binding-information) from the specification?

Reply:
CC @sbose78

Reply:
In subsequent PRs, yes! This PR is more of ensuring everything works well to a point and is well tested.

Reply:
sounds good!